### PR TITLE
MINIFI-216 - Allowing override of nifi.properties values

### PR DESCRIPTION
**Pull request by brosander**
Thank you for submitting a contribution to Apache NiFi - MiNiFi.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [X] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [X] Does your PR title start with MINIFI-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [X] Has your PR been rebased against the latest commit within the target branch (typically master)?

- [X] Is your initial contribution a single, squashed commit?

### For code changes:
- [X] Have you ensured that the full suite of tests is executed via mvn -Pcontrib-check clean install at the root nifi-minifi folder?
- [X] Have you written or updated unit tests to verify your changes?
- [X] - N/A - If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [X] - N/A - If applicable, have you updated the LICENSE file, including the main LICENSE file under minifi-assembly?
- [X] - N/A - If applicable, have you updated the NOTICE file, including the main NOTICE file found under minifi-assembly?

### For documentation related changes:
- [X] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: minifi-bootstrap/src/test/resources/MINIFI-216/config.yml [hunk 1]
```diff
@@ -0,0 +1,106 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
```
Comment (phrocker) [inline minifi-bootstrap/src/test/resources/MINIFI-216/config.yml]: +1 I think we should be able to share these test resources amongst cpp and java and have the same results or at least used in similar tests.

File: minifi-bootstrap/src/main/java/org/apache/nifi/minifi/bootstrap/util/ConfigTransformer.java [hunk 2]
```diff
@@ -704,7 +691,7 @@ protected static void addTextElement(final Element element, final String name, f
     }
 
     public static final String PROPERTIES_FILE_APACHE_2_0_LICENSE =
-            "# Licensed to the Apache Software Foundation (ASF) under one or more\n" +
+            " Licensed to the Apache Software Foundation (ASF) under one or more\n" +
```
Comment (phrocker) [inline minifi-bootstrap/src/main/java/org/apache/nifi/minifi/bootstrap/util/ConfigTransformer.java]: Why did you remove the comment markdown (octothorp -- #)?

Comment (brosander) [inline minifi-bootstrap/src/main/java/org/apache/nifi/minifi/bootstrap/util/ConfigTransformer.java]: @phrocker due to the change in the way the props file is written out.

[The store method](https://github.com/apache/nifi-minifi/pull/72/files#diff-a022622bcf128ab992d247f2a8694a0eR263) will prepend a # to the comment string.

**Comment by phrocker** [comment]
Built and tested locally. i ran a quick test flow that I've been using for testing.

**Comment by brosander** [comment]
Merging after +1 from @phrocker
